name: "autoware"
N: 20
integrator_step: 0.2
n_discs: 3

solver_settings:
  # solver: "acados" # acados or forces
  solver: "forces" # acados or forces
  floating_license: true
  enable_timeout: true
  init: 2 # 0 = cold start, 1 = centerer start, 2 = warm start with the selected primal variables
  use_sqp: false

debug_output: false

enable_output: true
control_frequency: 10
deceleration_at_infeasible: 1.0
max_obstacles: 12
robot_radius: 1.125
robot:
  length: 4.54
  width: 2.25
obstacle_radius: 0.4

linearized_constraints:
  add_halfspaces: 2 # For road constraints

road:
  two_way: false
  width: 12.0

shift_previous_solution_forward: false

contouring:
  num_segments: 8
  preview: 2.0 # [s], use 0.0 to disable preview

t-mpc:
  use_t-mpc++: true
  enable_constraints: true

probabilistic:
  enable: true
  risk: 0.05
  propagate_uncertainty: true

weights:
  goal: 10.
  velocity: 0.05 #0.5
  acceleration: 0.2
  angular_velocity: 0.25 #0.05
  reference_velocity: 3.0 #1.5
  contour: 0.1 #0.01
  lag: 0.2
  preview: 5.0
  slack: 1000.

# Copied from Prius: https://github.com/oscardegroot/lmpcc/blob/prius/lmpcc/cfg/PredictiveController_Prius.cfg
# Variable weights are not normalized
#
# weights:
#   goal: -1.
#   velocity: 0.05
#   acceleration: 0.2
#   angular_velocity: 0.5
#   reference_velocity: 5.0
#   contour: 0.03
#   lag: 0.2
#   slack: -1.

visualization:
  draw_every: 5 # stages
